converter/fluka/cards/scoring_card.py

```diff
@@ -1,42 +1,112 @@
 from dataclasses import dataclass, field
+from typing import Optional, Union
 
 from converter.fluka.cards.card import Card
-from converter.fluka.helper_parsers.scoring_parser import Scoring
+from converter.fluka.helper_parsers.detector_parser import Detector
+from converter.fluka.helper_parsers.scoring_parser import CustomFilter, ParticleFilter, Quantity, Scoring
 
 
-def handle_first_card(scoring: Scoring, output_unit: int = 21) -> str:
-    """Creates first Scoring card"""
+@dataclass
+class ScoringCardIndexCounter:
+    """Class representing counter for Fluka scoring cards"""
+
+    usrbin_counter: int = 0
+
+
+def handle_scoring_cards(output_unit: int, scoring: Scoring, counter: ScoringCardIndexCounter) -> str:
+    """Creates Scoring cards"""
+    output: list[str] = []
+    for quantity in scoring.quantities:
+        usrbin_card = handle_usrbin_scoring(scoring.detector, quantity, output_unit, counter)
+        output.append(usrbin_card)
+    return '\n'.join(output).strip()
+
+
+def short_name(name: str) -> str:
+    """Creates short name for fluka card"""
+    return f'{name[:10]}'
+
+
+def handle_usrbin_scoring(detector: Detector, quantity: Quantity, output_unit, counter: ScoringCardIndexCounter) -> str:
+    """Creates USRBIN card"""
+    output: list[Card] = []
     # temporary assumption
     binning_what = '10.0'
     # DOSE according to:
     # https://flukafiles.web.cern.ch/manual/chapters/particle_and_material_codes/particles_codes.html
-    particle_of_scoring = 'DOSE'
-    output_unit = str(output_unit * -1)
+    quantity_to_score = ''
+    try_auxscore = False
+    if quantity.keyword == 'Dose':
+        quantity_to_score = 'DOSE'
+        try_auxscore = True
+    elif quantity.keyword == 'Fluence':
+        if isinstance(quantity.scoring_filter, ParticleFilter):
+            # Apply particle from filter if fluency is used
+            quantity_to_score = quantity.scoring_filter.particle
+        elif quantity.keyword == 'Fluence' and isinstance(quantity.scoring_filter, CustomFilter):
+            quantity_to_score = 'ALL-PART'
+            try_auxscore = True
 
-    first_card = Card(codewd='USRBIN')
-    first_card.what = [
-        binning_what, particle_of_scoring, output_unit, scoring.detector.x_max, scoring.detector.y_max,
-        scoring.detector.z_max
-    ]
-    first_card.sdum = 'changeme'
+    if not quantity_to_score:
+        return ''
 
-    return first_card.__str__()
+    output_unit = str(output_unit * -1)
```

Comment:
Add a comment explaining what is going on here.
Its bad practice to modify parameter passed to the function, imagine what would happen if the function is executed twice.

Best would be to create new variables, say `output_unit_in_fluka_convention` or something meaningfluf